src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,117 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int threshold;
+    private int capacity;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (table == null || table.length == 0) {
+            table = new Node[DEFAULT_CAPACITY];
+        }
+        capacity = table.length;
+        threshold = (int) (capacity * LOAD_FACTOR);
+        if (size >= threshold) {
+            table = resize();
+        }
+        int index = Math.abs(hash(key) % capacity);
+        if (table[index] != null) {
+            Node<K, V> nextNode = table[index];
+            while (nextNode.next != null) {
+                if (Objects.equals(nextNode.key, key)) {
+                    nextNode.value = value;
+                    return;
+                }
+                nextNode = nextNode.next;
+            }
+            if (Objects.equals(nextNode.key, key)) {
+                nextNode.value = value;
+                return;
+            }
+            nextNode.next = putNodeInNullPosition(key, value);
+        }
+        if (table[index] == null) {
+            table[index] = putNodeInNullPosition(key, value);
+        }
```

Comment:
```suggestion
            Node<K, V> nextNode = table[index];
            while (nextNode.next != null) {
                if (Objects.equals(nextNode.key, key)) {
                    nextNode.value = value;
                    return;
                }
                nextNode = nextNode.next;
            }
            if (Objects.equals(nextNode.key, key)) {
                nextNode.value = value;
                return;
            }
            nextNode.next = putNodeInNullPosition(key, value);
```
Wouldn't the code work like that?